src/ray/raylet/node_manager.cc

```diff
@@ -542,6 +548,53 @@ ray::Status NodeManager::RegisterGcs() {
         "NodeManager.deadline_timer.print_event_loop_stats");
   }
 
+  if (RayConfig::instance().use_ray_syncer()) {
+    // Register resource manager and scheduler
+    ray_syncer_.Register(
+        /* component_id */ syncer::RayComponentId::RESOURCE_MANAGER,
+        /* reporter */ &cluster_resource_scheduler_->GetLocalResourceManager(),
+        /* receiver */ this,
+        /* upward_only */ false,
+        /* pull_from_reporter_interval_ms */
+        RayConfig::instance().raylet_report_resources_period_milliseconds());
+    ray_syncer_.Register(
+        /* component_id */ syncer::RayComponentId::SCHEDULER,
+        /* reporter */ this,
+        /* receiver */ nullptr,
+        /* upward_only */ true,
+        /* pull_from_reporter_interval_ms */
+        RayConfig::instance().raylet_report_loads_period_milliseconds());
+    // Register a commands channel.
+    // It's only used for GC right now.
+    ray_syncer_.Register(
+        /* component_id */ syncer::RayComponentId::COMMANDS,
+        /* reporter */ nullptr,
+        /* receiver */ this,
+        /* upward_only */ false,
+        /* pull_from_reporter_interval_ms */ 0);
+    periodical_runner_.RunFnPeriodically(
+        [this] {
+          // If plasma store is under high pressure, we should try to schedule a global
+          // gc.
+          auto triggered_by_global_gc = TryLocalGC();
+
+          if (triggered_by_global_gc) {
+            rpc::ResourcesData resources_data;
```

Comment:
I guess you want to separate GC sync with resource report sync? Should we stop reusing `rpc::ResourcesData`?

Reply:
Yes, we should. My plan here is to clean up this after rolling out. The reason is to simplify the code change in this PR.

Reply:
It includes: 1) move spcific logic of handling message into the right place; 2) update resource data to exclude necessary fields;